src/basic/fileio.c

```diff
@@ -780,6 +781,7 @@ static int merge_env_file_push(
 
         if (!env_name_is_valid(key)) {
                 log_error("%s:%u: invalid variable name \"%s\", ignoring.", strna(filename), line, key);
+                free(value);
```

Comment:
We have no support for skipping BOMs afaik, and parse_env_file does not validate variable name validity, so you end up with an entry called "<BOM>one". We probably should detect this better, but that's unrelated to my PR.